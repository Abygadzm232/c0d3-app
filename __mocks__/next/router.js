global.window = Object.create(window)
Object.defineProperty(global.window, 'location', {
  value: { pathname: '/not-root' } // make sure pathname isnt '/' by default
})
const push = jest.fn(path => (global.window.location['pathname'] = path))
export const useRouter = jest.fn().mockImplementation(() => {
  return {
    route: '/',
    pathname: '/',
    query: {},
    asPath: '/',
    push: push,
    replace: async () => true,
    reload: () => null,
    back: () => null
  }
})
