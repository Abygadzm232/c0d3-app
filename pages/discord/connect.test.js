import React from 'react'
import ConnectToDiscordPage from './connect'
import { render, fireEvent, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { MockedProvider } from '@apollo/client/testing'
import { useRouter } from 'next/router'

const { push, query } = useRouter()

jest.mock('next/link', () => {
  return ({ children }) => {
    return children
  }
})

import Link from 'next/link'

describe('connect to Discord page', () => {
  it('should render page', () => {})

  it('should direct to curriculum page when user declines to connect to Discord', () => {
    const { getByTestId } = render(<ConnectToDiscordPage />)

    const connectButton = getByTestId('button')
    fireEvent.click(connectButton)

    await waitFor(() => expect(push).toBeCalledWith('/curriculum'))
  })

  it('should start Discord auth flow when connect to discord button is clicked', () => {})
})
