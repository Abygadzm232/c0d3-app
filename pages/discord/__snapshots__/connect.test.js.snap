// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`connect to Discord page should render page 1`] = `
<div>
  <div
    class="row mt-5"
  >
    <div
      class="card shadow-sm col-sm-8 col-md-7 col-lg-6 col-xl-6 m-auto px-md-5 border-0"
    >
      <div
        class="card-body text-center pt-5 pb-5"
      >
        <h1
          class="card-title h2 font-weight-bold mb-5"
        >
          Connect to Discord
        </h1>
        <p
          class="card-text"
        />
        <div
          class="mt-3"
        >
          <p>
            Please
             
            <a
              class="button"
              href="https://discord.com/api/oauth2/authorize?client_id=845470281283665970&redirect_uri=http%3A%2F%2Flocalhost%3A3000%2Fapi%2Fauth%2Fcallback%2Fdiscord&response_type=code&scope=identify%20email%20guilds.join%20gdm.join"
            >
              connect your Discord account
            </a>
            , or create one if you don't already have one. Our students and mentors use Discord to communicate and help each other, give you feedback on your challenges, and sometimes do virtual hangouts.
          </p>
          <p>
            If you don't want to connect to Discord, then you won't get any value out of creating an account on C0D3.com. Feel free to browse our lessons and study on your own!
          </p>
          <p>
            Also, if you would like to share your reasons for not using Discord, we'd love to hear about it! Your thoughts will help us make more informed decisions about our community platform and if the reasons are compelling enough, may inspire us to switch to an alternative.
          </p>
        </div>
        <button
          class="btn btn-primary btn-lg btn-block mb-3"
          type="button"
        >
          Connect Now
        </button>
        <a
          class=""
          href="/curriculum"
        >
          No thanks, I'll study on my own. 
        </a>
      </div>
    </div>
  </div>
</div>
`;
